packages/tupaia-web/src/ModalRoutes.tsx

```diff
@@ -42,10 +42,15 @@ export const ModalRoutes = () => {
   const { hash, closeModal } = useModal();
 
   const modal = hash as typeof MODAL_ROUTES[keyof typeof MODAL_ROUTES];
+  useEffect(() => {
+    if (modal === null) {
+      gaEvent('User', 'Close Modal');
+    }
+    gaEvent('User', modal, 'Open Modal');
+  }, [modal]);
```

Comment:
I wonder if this will have the effect of running on mount as well? If so, we probably don't want that